glean-core/src/metrics/labeled.rs

```diff
@@ -184,32 +101,82 @@ where
     ///
     /// Labels must be `snake_case` and less than 30 characters.
     /// If an invalid label is used, the metric will be recorded in the special `OTHER_LABEL` label.
-    pub fn get<'a, G: Into<Option<&'a Glean>>>(&mut self, glean: G, label: &str) -> T {
-        // We have 3 scenarios to consider:
+    pub fn get(&mut self, label: &str) -> T {
+        // We have 2 scenarios to consider:
         // * Static labels. No database access needed. We just look at what is in memory.
-        // * Dynamic labels, initialized Glean. We look up in the database all previously stored
-        //   labels in order to keep a maximum of allowed labels.
-        // * Dynamic labels, uninitialized Glean. We ignore potentially previously stored labels
-        //   and just take what the user passed in.
-        //   This potentially allows creating more than `MAX_LABELS` labels, but only before Glean
-        //   is initialized.
-        //   This behavior is not publicly documented and should not be abused/depended upon.
-        // The last case is buggy behavior, tracked in bug 1588451.
-        let glean = glean.into();
-        let label = match (&self.labels, glean) {
-            (Some(_), _) => self.static_label(label),
-            (None, Some(glean)) => self.dynamic_label(glean, label),
-            (None, None) => {
-                if let Ok(label) = validate_label(label) {
-                    label
-                } else {
-                    // We can't report any error, as we don't have a Glean instance.
-                    OTHER_LABEL
-                }
+        // * Dynamic labels. We look up in the database all previously stored
+        //   labels in order to keep a maximum of allowed labels. This is done later
+        //   when the specific metric is actually recorded, when we are guaranteed to have
+        //   an initialized Glean object.
+        match self.labels {
+            Some(_) => {
+                let label = self.static_label(label).to_string();
+                self.new_metric_with_name(format!("{}/{}", self.submetric.meta().name, label))
             }
-        };
-        let label = format!("{}/{}", self.submetric.meta().name, label);
+            None => self.new_metric_with_dynamic_label(label.to_string()),
+        }
+    }
+}
 
-        self.new_metric_with_name(label)
+/// Correct a dynamic label
```

Comment:
nit: "Validates a dynamic label"?

Reply:
I changed the word because it really changes it, not just reporting when it's wrong.

Maybe "Validates a dynamic label, changing it to __other__ if it's invalid." ??

Reply:
Yes, good idea, I like your new phrasing more!